app/src/org/commcare/views/media/MediaLayout.java

```diff
@@ -342,129 +232,66 @@ private View setupImage(String imageURI, String bigImageURI) {
                     } else {
                         mImageView.setImageBitmap(b);
                     }
-                    mImageView.setId(IMAGE_VIEW_ID);
-                    mediaPane = mImageView;
+                    mImageView.setVisibility(VISIBLE);
                 }
             } else {
                 // An error hasn't been logged. We should have an image, but the file doesn't
                 // exist.
-                mediaPane = getMissingMediaView(imageURI,
-                        StringUtils.getStringRobust(getContext(), R.string.video_download_prompt),
-                        true);
+                showMissingMediaView(imageURI,
+                        StringUtils.getStringRobust(getContext(), R.string.image_download_prompt),
+                        true,
+                        () -> {
+                            hideMissingMediaView();
+                            setupImage(imageURI, bigImageURI);
+                        });
             }
         } catch (InvalidReferenceException e) {
             Log.e(TAG, "image invalid reference exception");
```

Comment:
Same, do we need to show something here as well?

Reply:
we can follow the same pattern as videoView.